nifi-nar-bundles/nifi-oauth-bundle/nifi-oauth/src/main/java/org/apache/nifi/oauth/AbstractOAuthControllerService.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.oauth;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+
+
+public abstract class AbstractOAuthControllerService
+    extends AbstractControllerService implements OAuth2ClientService {
+
+    protected String accessToken = null;
+    protected String refreshToken = null;
+    protected String tokenType = null;
+    protected long expiresIn = -1;
+    protected long expiresTime = -1;
+    protected long lastResponseTimestamp = -1;
+    protected Map<String, String> extraHeaders = new HashMap<String, String>();
+    protected String authUrl = null;
+    protected long expireTimeSafetyNetSeconds = -1;
+    protected String accessTokenRespName = null;
+    protected String expireTimeRespName = null;
+    protected String expireInRespName = null;
+    protected String tokenTypeRespName = null;
+    protected String scopeRespName = null;
+
+    public static final PropertyDescriptor AUTH_SERVER_URL = new PropertyDescriptor
+            .Builder().name("OAuth2 Authorization Server URL")
```

Comment:
The `name` should avoid special characters and spaces for serialization purposes. Suggest `.name("oauth2_authorization_server_url")`.

Reply:
Good point I will change the name and enable expression language by default